change-validation/src/main/scala/com/normation/plugins/changevalidation/CurrentUser.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ *************************************************************************************
+ * Copyright 2023 Normation SAS
+ *************************************************************************************
+ *
+ * This file is part of Rudder.
+ *
+ * Rudder is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In accordance with the terms of section 7 (7. Additional Terms.) of
+ * the GNU General Public License version 3, the copyright holders add
+ * the following Additional permissions:
+ * Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+ * Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+ * Public License version 3, when you create a Related Module, this
+ * Related Module is not considered as a part of the work and may be
+ * distributed under the license agreement of your choice.
+ * A "Related Module" means a set of sources files including their
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package com.normation.plugins.changevalidation
+
+import com.normation.rudder.AuthorizationType
+import com.normation.rudder.Rights
+import com.normation.rudder.web.services.RudderUserDetail
+import net.liftweb.http.SessionVar
+import org.springframework.security.core.context.SecurityContextHolder
+
+/**
+  * Redefine the Rights methods as in Rudder, constructor allows to provide RudderUserDetail
+  */
+class CurrentUser protected (get: => Option[RudderUserDetail]) extends SessionVar[Option[RudderUserDetail]](get) {
```

Comment:
You absolutly don't want to define these kind of security/authorization thing at several place. If it exists in Rudder, you must not ever define it elsewhere, it's a sure path toward inconsistancy and security hole. 

If you are missing something with Current user: 
- either you change it in Rudder, and ensure that everything still work as expected, 
- or you use adapt for test (if it's only a proble of test). 

Plus: Rudder `CurrentUser` has a `set` method, so you can `CurrentUser.set(Some(userDetails))` if needed

Reply:
:+1: I didn't think the `set` method existed, it simplifies a lot of things ! (morale : sometimes mutability does, and we need it sometimes) :smile:

Now I only have to wait for this PR to make the `set` method available : https://github.com/Normation/rudder/pull/5267

Reply:
Now that that one PR is merged I could not do the test without this other one :smile:  : https://github.com/Normation/rudder/pull/5289 (`set`-ting the user is not possible at top-level, we need to do it in each yaml test so it needs a change in the test trait in Rudder)

Reply:
A simpler and better solution : rely on the `UserService`, which allows indeed to have a user with the correct rights in tests : https://github.com/Normation/rudder-plugins/pull/618/commits/9d45951ef7ded84ca44692a0fd84f2c7aa87f8cd

With that last commit I should have addressed all issues for this PR...